library/kani_macros/src/lib.rs

```diff
@@ -51,6 +56,40 @@ pub fn proof(attr: TokenStream, item: TokenStream) -> TokenStream {
     // )
 }
 
+#[cfg(not(kani))]
+#[proc_macro_attribute]
+pub fn async_proof(attr: TokenStream, item: TokenStream) -> TokenStream {
+    proof(attr, item) // Treat like a normal #[kani::proof] if Kani is not active
+}
+
+#[cfg(kani)]
+#[proc_macro_attribute]
+pub fn async_proof(attr: TokenStream, item: TokenStream) -> TokenStream {
+    assert!(attr.to_string().is_empty(), "#[kani::async_proof] does not take any arguments");
```

Comment:
This is true for now, but will probably not be true in the future if we want to control which executor we use

Reply:
Agreed. I'll mention that this may change in the future.